File: fsharp/sum-of-multiples/SumOfMultiples.fs
module SumOfMultiples

let sum (numbers: int list) (upperBound: int): int = 
    let betterNumbers = numbers |> List.filter ((<) 0)
    if betterNumbers.Length = 0 then 
        0
    else
        betterNumbers 
        |> Seq.collect (fun n -> [n..n..(upperBound-1)]) 
        |> Seq.distinct 
        |> Seq.sum
        

File: elixir/sum-of-multiples/lib/sum_of_multiples.ex
defmodule SumOfMultiples do
  @doc """
  Adds up all numbers from 1 to a given end number that are multiples of the factors provided.
  """
  @spec to(non_neg_integer, [non_neg_integer]) :: non_neg_integer
  def to(_limit, []), do: 0
  def to(_limit, [0]), do: 0
  def to(limit, factors) do
    factors
    |> Enum.filter(fn n -> n > 0 end)
    |> Enum.flat_map(fn n -> Enum.to_list(n..(limit-1)//n) end)
    |> Enum.uniq()
    |> Enum.sum()
  end
end
